inst/include/demographic_projection.hpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "types.hpp"
+
+template <typename real_type>
```

Comment:
later on, as this stuff grows we might use namespaces to try and keep it under control (particularly to make clear which bits are "science" code and which bits are the framework)